analysis/src/Hint.ml

```diff
@@ -119,4 +119,60 @@ let inlay ~path ~pos ~maxLength ~debug =
                | Some value ->
                  if String.length label > value then None else Some result
                | None -> Some result)
-             | None -> None)))
\ No newline at end of file
+             | None -> None)))
+
+let codeLens ~path ~debug =
+  let lenses = ref [] in
+  let push loc =
+    let range = Utils.cmtLocToRange loc in
+    lenses := range :: !lenses
+  in
+  (* Code lenses are only emitted for functions right now. So look for value bindings that are functions,
+     and use the loc of the value binding itself so we can look up the full function type for our code lens. *)
+  let value_binding (iterator : Ast_iterator.iterator)
+      (vb : Parsetree.value_binding) =
+    (match vb with
+    | {
+     pvb_pat = {ppat_desc = Ppat_var _; ppat_loc};
+     pvb_expr = {pexp_desc = Pexp_fun _};
+    } ->
```

Comment:
This pattern matches the value binding to target 2 things:
1. `Ppat_var` means it's a variable, like `let whatever`. Assignments can also be destructures `let {something} = ....`, but this ensures we only target variables.
2. `Pexp_fun` means it's a function definition.

`pvb_pat` is the _pattern_, which is essentially the left hand side of `let whatever = something`. `pvb_expr` is the _expression_ being assigned, so the right hand side after `=`. We ensure that it is indeed a variable being assigned `Ppat_var`, and that it's being assigned a function definition `Pexp_fun`.